### ISSUE TITLE
New rule: restrict use of aria-description and aria-braille*

### ISSUE THREAD

**Issue by straker** [dequelabs/axe-core#3955]
Axe-core 4.7 will add the `aria-description` and all `aria-brille*` attributes as valid ARIA attributes. Once that is done, we'll need to create a new rule (or two) that limits their use to elements that have an ARIA role and an accessible name.

**Comment by WilcoFiers** [comment]
I looked at this more closely. I think the right way to do this would be for ARIA to prohibit these attributes for `none` / `presentation` / `generic`. That covers the "role" scenario. I've opened an issue for it here: https://github.com/w3c/aria/issues/1988

What I think is left is a rule that checks that you can only use aria-braillelabel on elements with a non-empty accessible name, and that you can only use aria-brailleroledescription on elements with a non-empty aria-roledescription.

**Comment by padmavemulapati** [comment]
Verified on axe-core develop branch code base, 
for the test code snippet

```
<button id="fail1" aria-braillelabel="hello"></button>
<aside
  id="fail2"
  aria-roledescription=""
  aria-brailleroledescription="table of contents"></aside>

```

Seeing a new rule `aria-braille-equivalent`

```
description: "Ensure aria-braillelabel and aria-brailleroledescription have a non-braille equivalent"
help: "aria-braille attributes must have a non-braille equivalent"
helpUrl: "https://dequeuniversity.com/rules/axe/4.7/aria-braille-equivalent?application=axeAPI"
id: "aria-braille-equivalent"
impact: "serious"
nodes: Array(2)0: {any: Array(0), all: Array(1), none: Array(0), impact: 'serious', html: '<button id="fail1" aria-braillelabel="hello"></button>', …}
1: {any: Array(0), all: Array(1), none: Array(0), impact: 'serious', html: '<aside id="fail2" aria-roledescription="" aria-brailleroledescription="table of contents"></aside>', …}
tags:  ['cat.aria', 'wcag2a', 'wcag412', 'EN-301-549', 'EN-9.4.1.2']

```



![Image](https://github.com/dequelabs/axe-core/assets/47409579/c2a04783-5e8f-43f6-a6b6-a25e7d22a69e)
